sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/common/FileBasedIOITHelper.java

```diff
@@ -65,6 +68,13 @@ public void processElement(ProcessContext c) {
     }
   }
 
+  public static void publishToBigQuery(
+      Collection<NamedTestResult> results, String bigQueryDataset, String bigQueryTable) {
+    BigQueryResultsPublisher publisher =
```

Comment:
There already is a method that suits this case. Could you simplify it even further with: 

`BigQueryResultsPublisher.create().publish() `

using the `BigQueryResutsPublisher.publish(Collection<? extends TestResult> results, String tableName)` method that can accept collection. ([link](https://github.com/apache/beam/blob/67c8e7e6e2ded96b0d854f3b0a43e3ae83c1df6a/sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/BigQueryResultsPublisher.java#L54))

Reply:
For the future record: we discussed a bit about extracting a base class that would encapsulate collecting + publishing metrics. Such class could be then extended by particular test cases so that only relevant code is there. This would drastically reduce duplication. 

This is way out of scope of this PR. I created an issue for this: https://issues.apache.org/jira/browse/BEAM-6662

Reply:
Refactored reporting to use`BigQueryResutsPublisher.publish(Collection<? extends TestResult> results, String tableName)` method